### PR TITLE
Tentative Cygwin support for dll

### PR DESCRIPTION
**Pull request by KoichiYasuoka**
Quick hack to support Cygwin-based dll.

### CODE REVIEW

File: setuptools_rust/build.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ def build_extension(self, ext):
                     except subprocess.CalledProcessError as e:
                         pass
 
-            if executable:
+            if executable or sys.platform == "win32" or sys.platform == "cygwin":
```
Comment (kngwyu) [inline setuptools_rust/build.py]: You mean we need `chmod` for dlls?

**Comment by kngwyu** [comment]
Thanks!

File: setuptools_rust/build.py (hunk 1 above)
Comment (KoichiYasuoka) [reply setuptools_rust/build.py]: Yes. For example, [tokenizers.dll](https://pypi.org/project/tokenizers/) needs to `chmod u+x` when installing on Cygwin.

Comment (kngwyu) [reply setuptools_rust/build.py]: - How about doing this hack only for cygwin? Is it also required for win32 platform?
- Please add a comment that explains why cygwin needs it.

Comment (KoichiYasuoka) [reply setuptools_rust/build.py]: This is for "cross-compile" between win32 and cygwin. Now I'm trying to compile PyO3 dll's for win32 with cygwin gcc, and then make wheels for win32 with setuptools_rust. In such cases, I need `sys.platform == "win32"` here. I agree that it is very rare case now.

Comment (KoichiYasuoka) [reply setuptools_rust/build.py]: I think that MSVC always outputs dll's with `-rwxr-x---` modes, so they don't need `chmod` here explicitly.

**Comment by kngwyu** [comment]
LGTM, thank you!

**Pull request merged**